src/matlab2tikz.m

```diff
@@ -2460,34 +2480,29 @@ case size(cData,2) % specific y-data is given
         needsPointmeta = any(xor(isnan(dz(:)), isnan(CData(:)))) ...
             || any(abs(CData(:) - dz(:)) > 1.0e-10);
         if needsPointmeta
-            % Get color map.
-            formatType = 'table[row sep=crcr,header=false,point meta=\thisrowno{3}]';
-            color      = CData(:);
+            columnNames = [columnNames, 'c'];
+            formatType  = 'table[row sep=crcr, point meta=\thisrow{c}]';
+            color       = CData(:);
         else
-            formatType = 'table[row sep=crcr,header=false,point meta=\thisrowno{2}]';
-            color = '';
+            formatType = 'table[row sep=crcr, point meta=\thisrow{z}]';
+            color      = [];
         end
     end
-    data = applyHgTransform(m2t, [dx(:), dy(:), dz(:)]);
-
+    data = [data, color];
+        
     % Add mesh/rows=<num rows> for specifying the row data instead of empty
     % lines in the data list below. This makes it possible to reduce the
     % data writing to one single sprintf() call.
     opts = opts_add(opts,'mesh/rows',sprintf('%d', numrows));
 
-    opts = opts_print(m2t, opts, ',\n');
-    str = [str, sprintf(['\n\\addplot3[%%\n%s,\n', opts ,']'], s.plotType)];
+    str = [str, sprintf('\n\\%s[%%\n%s,\n%s]', plotCmd, s.plotType, opts_print(m2t, opts, ','))];
```

Comment:
may be an additional comment: `% finally print depending on the present plot command`